rpc/handler.go

```diff
@@ -193,6 +185,15 @@ func (h *handler) handleBatch(msgs []*jsonrpcMessage) {
 	if len(calls) == 0 {
 		return
 	}
+
+	if len(calls) > h.batchRequestLimit && h.batchRequestLimit != 0 {
```

Comment:
Why was this moved -- if we want resiliency, it's better to handle batch too large before handling `non-call messages`